src/main/scala/com/redislabs/provider/redis/redisFunctions.scala

```diff
@@ -442,6 +455,33 @@ object RedisContext extends Serializable {
       }
   }
 
+  /**
+    * @param hashes hashName: map of k/vs to be saved in the target host
+    * @param ttl time to live
+    */
+  def setByteHash(hashes: Iterator[(Array[Byte], Map[Array[Byte], Array[Byte]])],
+              ttl: Int,
+              redisConfig: RedisConfig,
+              readWriteConfig: ReadWriteConfig) {
+    implicit val rwConf: ReadWriteConfig = readWriteConfig
+
+    hashes
+      .map { case (key, hashFields) =>
+        (redisConfig.getHost(key), (key, hashFields))
+      }
+      .toArray
```

Comment:
We take in an `Iterator`, but here we are materializing it in one go with `.toArray`. This could cause OOM on the spark driver. Can we work in batches using `.grouped(SOME_BATCH_SIZE)` and do all the subsequent things per batch?

Reply:
Hi @mayankasthana ,
Thank you for the comment, good point. We do grouping when writing dataframe in `RedisSourceRelation`, but it's missing in all RDD write functions. I'll fix that.